eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -19,18 +19,33 @@ function Get-PurgeableGroupResources {
 }
 
 function Get-PurgeableResources {
+    $purgeableResources = @()
```

Comment:
You don't need to collect them to then return them. Anything written to the pipeline is returned. If only 0 or 1 is, that's why I wrap the function in a `@()` call which coerces to a collection. Why the change here?

Reply:
Or was the idea merely to make sure you return a collection and not a collection of collections? You could do this, or just use `Write-Output $mycollection` which, by default, writes each item in the collection to the pipeline.

The advantage of doing this is that any subsequent cmdlets that handles pipeline output can start processing right away instead of waiting for all output.

```powershell
function foo { $a = 1..5; write-output $a; write-host "break"; $b = 6..10; write-output $b }
filter bar { write-host "Got $_" } # equivalent to function bar -process {}
foo | bar
```
```
Got 1
Got 2
Got 3
Got 4
Got 5
break
Got 6
Got 7
Got 8
Got 9
Got 10
```

Reply:
And if you'd still rather collect into a collection then return, may as well remove the "and coerce into ..." comments and get rid of the `@()` wrapper around `Get-PurgeableResources`. At the very least, you'd still be returning an empty collection from this PR's changes.

Still the checks below of `$myCollectionVar -and $myCollectionVar.Count` are unnecessary. Just `$myCollectionVar` is sufficient.

Reply:
removed the coerce operators.